paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -935,6 +939,34 @@ def deploy_group_is_set_to_notify(self, notify_type):
             self.deploy_info, self.deploy_group, notify_type
         )
 
+    def __build_rollback_audit_details(
+        self, rollback_type: RollbackTypes
+    ) -> Dict[str, str]:
+        return {
+            "rolled_back_from": self.commit,
+            "rolled_back_to": self.old_git_sha,
+            "rollback_type": rollback_type.value,
+            "deploy_group": self.deploy_group,
+        }
+
+    def log_slo_rollback(self) -> None:
+        _log_audit(
```

Comment:
How bullet proof is this function call? Under no circumstances should a failure here block a deploy or crash a rollback or anything bad.

The user experience is particularly bad when things like this happen because the slack interface just "stops working"

Reply:
afaik, it's pretty bullet-proof (it's using some pretty battle-tested logging code underneath) and I can't think of a way to make my helper function throw an exception (that wouldn't be caught by mypy before a release) - if we want to be extra-safe, we could guard this with a catch-all try-except.